lib/oppen.rb

```diff
@@ -12,13 +12,16 @@ module Oppen
   # Entry point of the pretty printer.
   #
   # @param config [Config]
+  # @param genspace [String/Proc] space generator: can be a string, a lambda, a block, ...
   # @param margin [Integer] maximum line width desired
   # @param new_line [String] the delimiter between lines
+  # @param out [Buffer] should have a write method
   # @param tokens [Array[Token]] the list of tokens to be printed
   #
   # @return [StringIO] output of the pretty printer
-  def self.print(config: Config.oppen, margin: 80, new_line: "\n", tokens: [])
-    printer = Printer.new margin, new_line, config
+  def self.print(config: Config.oppen, genspace: ->(n) { ' ' * n },
```

Comment:
rename `genspace` to `space`, and default it to `' '`. In the docs you can explain that you can also pass a callable like a lambda, and how we would translate internally that var into a lambda.